src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala

```diff
@@ -201,23 +201,17 @@ class EntityService(rawlsDAO: RawlsDAO, importServiceDAO: ImportServiceDAO, goog
     // The async path only supports upsert TSVs, not update TSVs. Import Service only operates in upsert mode;
     // if we tried to send an update TSV through it, it would lose its "update-only" restriction and become an upsert.
     if (isAsync) {
-      if (!isUpsert) {
-        // until Rawls & Import Service implement update-only TSVs, throw a BadRequest if the user requested async
-        // but sent an update TSV.
-        Future(RequestCompleteWithErrorReport(BadRequest, "Update-only TSVs cannot use async mode"))
-      } else {
-        asyncUpsert(workspaceNamespace, workspaceName, rawlsCalls, userInfo).recover {
-          case e: Exception =>
-            RequestCompleteWithErrorReport(InternalServerError, "Unexpected error during async TSV import", e)
-        }
+      asyncUpsert(workspaceNamespace, workspaceName, isUpsert, rawlsCalls, userInfo).recover {
```

Comment:
Should we change code comments on lines 201-202?

Reply:
yes, we should! Good catch; removed